### PR TITLE
ENH: Update Doxygen to 1.8.15

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

**Comment by hjmjohnson** [comment]
http://review.source.kitware.com/#/c/23777/

Matt McCormick
Uploaded patch set 1.Sep 29 8:36 AM

Kitware Build Robot
Patch Set 1: Verified+1 Build Successful: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=23777-1&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Sep 29 10:05 AM

Matt McCormick
Uploaded patch set 2.Oct 7 9:19 PM

Kitware Build Robot
Patch Set 2: Verified+1 Build Successful: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=23777-2&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 7 10:48 PM

Matt McCormick
Uploaded patch set 3: Patch Set 2 was rebased.Oct 22 5:25 PM

Kitware Build Robot
Patch Set 3: Verified+1 Build Successful: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=23777-3&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 22 5:54 PM

Hans J. Johnson
Patch Set 3: Does this still need to be a WIP?Oct 24 1:30 PM

Matt McCormick
Patch Set 3: > Still a WIP? Yes, there are hundreds of warnings. I doubt the output is correct. If someone else wants to run with it, please do.Oct 24 1:41 PM

Jon Haitz Legarreta Gorroño
Oct 24 1:53 PM
↩
Patch Set 3:

(2 comments)

The dash stuff is always hard to visaully spot, so nice !

BTW, I'm wondering whether it is necessary/useful for Doxygen to track the markdown files as well, if at all that is possible, and if the change in this topic was at all due to some Doxygen-related warning on that file.

Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.h
Line 49:
Last 'e' may be slipped in while typing?

Line 62:
Not sure whether this is the correct ending, or else whether \end{verbatim} should be used.


Dzenan Zukic
Oct 24 1:53 PM
↩
Patch Set 3: Code-Review+1

(2 comments)

Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.h
Line 64:
is the second asterisk intentional?

Utilities/Doxygen/doxygen.config.in
Line 299:
Should we put here
in=C++
See paragraph above for explanation.


Matt McCormick
Oct 29 4:19 PM
↩
Patch Set 3:

(4 comments)

Thanks for the reviews!

Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.h
Line 49:
Nice catch!! Thanks!

Line 62:
Yes, we want \endverbatim here because it is a Doxygen Command:

  https://www.stack.nl/~dimitri/doxygen/manual/commands.html

Line 64:
Removed.

Utilities/Doxygen/doxygen.config.in
Line 299:
In general, I don't think the template files will be parsable by Doxygen because of the template content.


Matt McCormick
Uploaded patch set 4.Oct 29 4:20 PM

Matt McCormick
Patch Set 4: Latest build: https://open.cdash.org/viewBuildError.php?type=1&buildid=5604435Oct 29 5:59 PM

Kitware Build Robot
Patch Set 4: Verified+1 Build Successful: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=23777-4&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 29 6:04 PM

Jon Hait... Gorroño
Patch Set 3: +1Oct 29 7:38 PM

Dzenan Zukic
Patch Set 4: Code-Review+1 Updating is good. But, are there now less or more errors? :DOct 30 10:14 AM

**Comment by hjmjohnson** [comment]
Accidentally deleted the branch in my account.

**Comment by dzenanz** [comment]
Should this be merged? Or abandoned?

**Comment by thewtex** [comment]
This needs need more work to be merged, but it should not be abandoned.

**Comment by thewtex** [comment]
@stalebot this is still on our todo list.

File: Modules/Core/Transform/include/itkAzimuthElevationToCartesianTransform.h [hunk 1]
```diff
@@ -67,12 +71,12 @@ namespace itk
  * over the scalar type?  Perhaps it should always be double, unless
  * there's a compatibility problem with the Point class.
  *
- * \todo  Derive this class from a yet undefined TransformBase class.
- *        Currently, this class derives from AffineTransform, although
- *        it is not an affine transform.
+ * \todo Derive this class from a yet undefined TransformBase class.
```
Comment (dzenanz) [inline Modules/Core/Transform/include/itkAzimuthElevationToCartesianTransform.h]: `TransformBase` now exists, this sentence could indicate that.

**Comment by dzenanz** [comment]
[`MultiThreader`](https://github.com/InsightSoftwareConsortium/ITK/blob/release-4.13/Modules/Core/Common/include/itkMultiThreader.h) in v4 and its successor [`PlatformMultiThreader`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkPlatformMultiThreader.h) in v5 are missing from doxygen documentation. For example, it is missing from inheritance diagram in [MultiThreaderBase](https://itk.org/Doxygen/html/classitk_1_1MultiThreaderBase.html). Can you figure out why?

**Comment by hjmjohnson** [comment]
> [`PlatformMultiThreader`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkPlatformMultiThreader.h)

I don't want to complicate this PR with trying to address that issue.  It could be that a new PR needs to be created to fix the doxygen code:  

This seems to be copy-n-pasted incorrectly:

https://github.com/InsightSoftwareConsortium/ITK/blob/ce68cc01ce9b27a33c5e9972d31075975a4025e3/Modules/Core/Common/include/itkPlatformMultiThreader.h#L37


it should be
```c++
/** \class PlatformMultiThreader
```

**Comment by dzenanz** [comment]
Suggested fix in #989.

**Pull request merged**